src/src/com/microsoft/aad/adal/AuthenticationContext.java

```diff
@@ -1440,6 +1440,15 @@ private AuthenticationResult localFlow(CallbackHandler callbackHandle,
             } catch (AuthenticationException authenticationException) {
                 callbackHandle.onError(authenticationException);
                 return null;
+            } catch (HttpResponseException httpResponseException) {
+                // Return back the http error for silent request
+                // Try interactive flow for non-silent request.
+                if (request.isSilent()) {
+                    final AuthenticationException authenticationException = new AuthenticationException(ADALError.AUTH_REFRESH_FAILED_PROMPT_NOT_ALLOWED, 
```

Comment:
And overall I'm just confused that we want to return AUTH_REFRESH_FAILED_PROMPT_NOT_ALLOWED always? what is the point to have other error codes?